systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java

```diff
@@ -322,6 +326,152 @@ void testKafkaPodPendingDueToRack(ExtensionContext extensionContext) {
         KafkaUtils.waitForKafkaDeletion(namespaceName, clusterName);
     }
 
+    /**
+     * @description This test case examines the behavior of Kafka rollers under various node pool configurations, focusing on how rolling updates are triggered or prevented in different scenarios.
+     *
+     * @steps
+     *  1. - Initialize test environment and create node pools with different roles (mixed, broker, controller) and varying replica counts.
+     *     - Create Kafka instance with a single replica and deploy mixed node pool 'A' with 1 replica.
+     *  2. - Deploy two additional broker node pools 'B1' and 'B2' with 1 and 2 replicas, respectively.
+     *     - Verify that the creation of these broker nodes does not trigger a rolling update in the existing node pool 'A'.
+     *  3. - In KRaft mode, deploy two controller node pools 'C1' and 'C2' with 2 and 1 replicas, respectively.
+     *     - Expect a rolling update for mixed-role and broker-role nodes, but not for the controller nodes initially.
+     *  4. - After creating controller node pools, verify rolling updates across different node pools as they adjust to the new configuration.
+     *     - Check that the controller nodes roll only when their specific pool is modified.
+     *  5. - Change controller-only configuration and verify that only mixed and controller node pools are affected, while broker nodes remain unchanged.
+     *  6. - Modify broker-only configuration and observe that only mixed-role and broker-role node pools roll, leaving controller-role nodes unaffected.
+     *
+     * @usecase
+     *  - kafka-node-pool-rolling-update
+     *  - kafka-roller
+     *  - kafka-configuration-management
+     */
```

Comment:
Not a strong opinion if all ST SMEs are fine with it. I'm not sure this is optimal ...
* There are several open issues for scaling the controllers (adding a new controller or mixed node pool is scaling controllers)
* Might be worth separating this into two tests => one for dedicated controllers and one for mixed nodes.
* I wonder if separating the testing of adding / removing node pools into separate tests would be worth it as well to create something what is more simple and easier to maintain

Reply:
Hmmm and what about having:
1. Test case - scale controllers-only nodes 
2. Test case - scale mixed-only nodes
3. Test case - current test case doing both (but removing configuration changes I think to make it simpler)
4. Test case - changing the controller-only and broker-only configuration  (this could be I think one test case)
? :)

Reply:
Yeah that makes sense. I guess it will be more clear what we are testing. At least we will not think every time about what is tested or not based on env variables.

Reply:
Well, yes ... but ...
* Some of them are not working well because of known limitations ... such as 1 and 2
* Rolling updates of mixed nodes are useful as well
* Not sure what exactly you mean with 3 => scaling both mixed and controller nodes together? That seems to make little sense as you would normally use one or the other.

Reply:
as you all mentioned, it seems to me reasonable to have following tests: 
- rolling and config changes related to controllers when there is controller only role (implemented here)
- rolling and config changes related to controllers when there are mixed roles (implemented here) 
- adding removing node pools (right now broker only roles) (implemented here https://github.com/strimzi/strimzi-kafka-operator/pull/9428)  

mentioned things (test improvements) like scaling controllers (adding controller) node pool or additional checks for correct configuration propagation can be added later accordingly.